### ISSUE TITLE
DID Resolution Result for DID URL Dereferencing

### ISSUE THREAD

**Issue by clehner** [w3c/did-resolution#69]
[DID Resolution Result](https://w3c-ccg.github.io/did-resolution/#did-resolution-result) says:
> This section defines a data structure that represents the result of the algorithms described in § 3. Resolving a DID and § 4. Dereferencing a DID URL. A DID resolution result contains a DID document as well as DID resolution metadata and DID document metadata.

The result of *§ 4. Dereferencing a DID URL* might not be a DID document but could be something else, e.g. a verification method ([Example 8](https://w3c-ccg.github.io/did-resolution/#example-8)), or a output service endpoint URL ([Example 10](https://w3c-ccg.github.io/did-resolution/#example-10)). DID Core (in [§ 7.2 DID URL Dereferencing.](https://www.w3.org/TR/did-core/#did-url-dereferencing)) says that `contentStream`, the resource part of the return value of DID URL dereferencing, "MAY be a resource such as a DID document that is serializable in one of the conformant representations, a Verification Method, a service, or any other resource format that can be identified via a Media Type and obtained through the resolution process".

Should the DID Resolution Result section, and the JSON-LD context file, be updated to support DID URL dereferencing results other than DID documents?

In DID Test Suite there are examples of outputs of DID URL dereferencing, similar to the outputs for DID resolution, which perhaps could be useful here. In these, the `contentStream` property is a string (even if the content is a DID Document or JSON/JSON-LD object), `dereferencingMetadata` is an object, like `didResolutionMetadata`; and `contentMetadata` is an object, like `didDocumentMetadata`. e.g.:
- https://github.com/w3c/did-test-suite/blob/1b86f694ad40e0b61472f02e8cd06a200ac2b4f9/packages/did-core-test-server/suites/implementations/dereferencer-polygon-ayanworks.json#L18-L24 DID document resolved with no parameters
- https://github.com/w3c/did-test-suite/blob/1b86f694ad40e0b61472f02e8cd06a200ac2b4f9/packages/did-core-test-server/suites/implementations/dereferencer-3-3box-labs.json#L19-L31 - DID document resolved with `versionTime`
- https://github.com/w3c/did-test-suite/blob/1b86f694ad40e0b61472f02e8cd06a200ac2b4f9/packages/did-core-test-server/suites/implementations/dereferencer-mattr.json#L107-L122 service+relativeRef returning a URL
- https://github.com/w3c/did-test-suite/blob/1b86f694ad40e0b61472f02e8cd06a200ac2b4f9/packages/did-core-test-server/suites/implementations/dereferencer-web-transmute.json#L19-L25 service+relativeRef returning a file
- https://github.com/w3c/did-test-suite/blob/1b86f694ad40e0b61472f02e8cd06a200ac2b4f9/packages/did-core-test-server/suites/implementations/dereferencer-spruce-key.json#L35-L41 verification method (JSON-LD)
- https://github.com/w3c/did-test-suite/blob/1b86f694ad40e0b61472f02e8cd06a200ac2b4f9/packages/did-core-test-server/suites/implementations/dereferencer-mattr.json#L55-L61 verification method (JSON)

**Comment by peacekeeper** [comment]
> Should the DID Resolution Result section, and the JSON-LD context file, be updated to support DID URL dereferencing results other than DID documents?

@clehner yes definitely. You correctly identified a gap where the DID Resolution spec is lagging behind DID Core a bit. This spec needs to better describe the output structures of DID URL dereferencing.

I wrote the part of the DID test suite that deals with resolving and dereferencing, and I agree this should also be reflected here. I'll try to work on this as soon as there's a bit of time.

**Comment by peacekeeper** [comment]
Coming back to this issue, there is now a section called "DID URL Dereferencing Result": https://w3c.github.io/did-resolution/#did-url-dereferencing-result

But that section probably needs a better definition and some more information about the data structure.

**Comment by pchampin** [comment]
This was discussed during the [did meeting on 23 January 2025](https://www.w3.org/2025/01/23-did-minutes.html#ffe3).

<details><summary><i>View the transcript</i></summary>

<h4><a href="https://github.com/w3c/did-resolution/issues/69" rel="noopener noreferrer">w3c/<wbr>did-resolution#69</a></h4>
<p><cite>manu:</cite> Not clear what a network is.  No examples.</p>

<hr /></details>

**Comment by wip-abramson** [comment]
> This was discussed during the [did meeting on 23 January 2025](https://www.w3.org/2025/01/23-did-minutes.html#ffe3).

Based on discussion, it was agreed that this issue has been addressed. Marking pending close.
